index.js

```diff
@@ -98,7 +98,7 @@ const register = (server, option) => {
         },
         handler(request, h) {
             request.cookieAuth.clear();
-            return h.redirect(resolveNext(request.query));
+            return h.redirect(`https://${config.auth0Domain}/v2/logout?returnTo=${encodeURI(request.server.info.uri + resolveNext(request.query))}`);
```

Comment:
This is great in general. But the problem with it is that Auth0 requires the value for `returnTo` to be on a whitelist of URLs, which we have to manually specify. That will break things because now when I visit `/logout?next=/faq`, for example, Auth0 will return an error until I add the FAQ page to the whitelist. I would prefer to have Auth0 always redirect back here to `/logout` (so that only `/logout` needs to be whitelisted) and then we can redirect the user to `next`.

To do this, we would need to implement something similar to what `bell` does with the `state` query parameter. Basically generate a value for a valid, previously generated `state`. If yes, redirect to `next`, otherwise redirect to Auth0.

Reply:
You talking about that part of `bell` https://github.com/venikman/bell/blob/master/lib/oauth.js#L202-L223 @sholladay  ?

Reply:
On the initial request to log in:
1. First they [create a nonce](https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L183)
2. Next they [save the nonce in a cookie](https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L211-L216) so it can be retrieved on future requests (only safe to trust because the cookie is encrypted)
3. Then they also send the nonce to Auth0:
    - https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L193
    - https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L217

On the callback request, where Auth0 redirects back to the login route:
1. First they [delete the cookie](https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L228) since it was only needed temporarily to remember the nonce between requests
2. Finally, they [compare the nonce](https://github.com/hapijs/bell/blob/16a491180231ae06fce84ded929d20a5ab6d7d3d/lib/oauth.js#L231-L233) that was saved in the cookie vs the one Auth0 sends back and only proceeds if they match

Frankly, it is pretty clever. It improves security because bell is able to be pretty certain that it is actually Auth0 who is sending the second request, instead of some hacker pretending to be the Auth0.

BTW, your fork is way behind upstream `bell`!